src/Form/LegacyConsumer/Commands/SetupFieldValidation.php

```diff
@@ -60,6 +61,10 @@ public function __invoke($hook)
      */
     public function validate(Field $field)
     {
+        if( ! $this->isFieldVisible( $field ) ) {
+            return;
+        }
+
```

Comment:
Short-circuiting the validation process if a field isn't visible might cause some issues. For example, a required File Upload field that only accepts image files is set to display when then donation amount is $25. The user can set the donation amount to $25, upload a .exe file, and then change the donation amount. Now the file upload field is hidden, and the donation continues even though the upload is invalid. Since hidden fields are still committed to the database, the file is uploaded successfully. (I'm not sure if hidden fields **should** get saved, but that's outside of the scope of the original issue.)

Note: This is related to #385, but that only applies when the field isn't required.

Reply:
I would say that hidden fields *should not* be persisted - they should be treated as empty. This needs to be addressed as well, but in the Form Field Manager add-on.

Reply:
@LMNTL I have created a separate issue in FFM to address this, see https://github.com/impress-org/give-form-field-manager/issues/391.

Reply:
@kjohnson That makes sense to me! Approving this.